src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -408,6 +424,9 @@ public async Task<INetworkPeer> ConnectAsync(IPEndPoint ipEndpoint)
             {
                 this.peerAddressManager.PeerAttempted(ipEndpoint, this.dateTimeProvider.GetUtcNow());
                 await peer.VersionHandshakeAsync(this.nodeLifetime.ApplicationStopping).ConfigureAwait(false);
+
+                // Set external address in self endpoint tracker to peer in version payload.
+                this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(peer.PeerVersion.AddressFrom, null);
```

Comment:
We actually said, that we want this to be hard of VersionHandshakeAsync

Reply:
Didn't understand this comment so will wait for a clarification.

Reply:
What I mean is we should remove these three lines because the we handle it elsewhere